src/native/eventpipe/ds-ipc-posix.c

```diff
@@ -58,29 +58,33 @@
 #define DS_EXIT_BLOCKING_PAL_SECTION
 #endif
 
+#ifndef __cplusplus
```

Comment:
Why is this conditional on the compilation language, rather than on some configuration parameter (for example: which runtime is being compiled)?  I don't think it's wrong, just curious.

Reply:
Just for simplicity, but we could move it down into ep-rt-config-coreclr and set a define that is then checked on all the allocator macros in the shim.

Reply:
I think that would be better. We always had troubles with whether the `.c` files should be compiled as `C` or `C++` and kept flipping it back and forth.

Reply:
sounds good, will introduce a define that can be set in the shim config controlling the use of allocator in the ipc sources.